ghcide/src/Development/IDE/GHC/Util.hs

```diff
@@ -318,4 +314,45 @@ traceAst lbl x
 #endif
             , "file://" ++ htmlDumpFileName]
 
+-- Should in `Development.IDE.GHC.Orphans`,
+-- leave it here to prevent cyclic module dependency
+#if !MIN_VERSION_ghc(8,10,0)
+instance Outputable SDoc where
+  ppr = id
+#endif
 
+-- | Print a GHC value by default `showSDocUnsafe`.
+--
+-- You may prefer `prettyPrint` unless you know what you are doing.
+--
+-- It internal using `unsafeGlobalDynFlags`.
+--
+-- `String` version of `showGhcWithUniques`.
+prettyPrintWithUniques :: Outputable a => a -> String
+prettyPrintWithUniques = showSDocUnsafe . ppr
+
+-- | Print a GHC value in `defaultUserStyle` without unique symbols.
+--
+-- This is the most common print utility, will print with a user friendly style like: `a_a4ME` as `a`.
+--
+-- It internal using `unsafeGlobalDynFlags`.
+--
+-- `String` version of `showGhc`.
+prettyPrint :: Outputable a => a -> String
+prettyPrint = printWithoutUniques
+
+-- | Print a GHC value by default `showSDocUnsafe`.
+--
+-- You may prefer `showGhc` unless you know what you are doing.
+--
+-- It internal using `unsafeGlobalDynFlags`.
+showGhcWithUniques :: Outputable a => a -> T.Text
+showGhcWithUniques = T.pack . showSDocUnsafe . ppr
+
+-- | Print a GHC value in `defaultUserStyle` without unique symbols.
+--
+-- This is the most common print utility, will print with a user friendly style like: `a_a4ME` as `a`.
+--
+-- It internal using `unsafeGlobalDynFlags`.
+showGhc :: Outputable a => a -> T.Text
```

Comment:
In the interests of keeping things simple, I think we could get away with just this function. Replacing `prettyPrint` just requires `show $ showGhc x` at the call site.

Also, name bikeshedding: how about `printOutputable`?

Reply:
I keeped `printOutputable :: Outputable a => a -> String` and `printOutputableText :: Outputable a => a -> Text`. I think we shouldn't pay for extra `O(n)` to `pack`  if `Text` is not required, and also, a lot of `pack $ printOutputable x` looks inconvenient.

It is only my personal aspect, I'm happy to revise them if you have different views:)

Reply:
I was thinking the other way around: just have the one that prints `Text` and let people `show` it if they need a `String`. `Text` by default is a better way anyway!

I think the cost of string conversions can be assumed to be totally trivial given the size of things we're talking about! I'm more concerned with the complexity of the codebase, so the fewer functions the better...